app/main/routes.py

```diff
@@ -0,0 +1,217 @@
+    """
+    Update BucketList.
+
+    Update a BucketList name.
+    """
+    bucketlist = BucketList.query.filter_by(bucketlist_id=list_id).first()
+
+    if not bucketlist or bucketlist.created_by != g.user.user_id:
+        return errors.not_found("The BucketList with the id: {0} doesn't"
+                                " exist.".format(list_id))
+
+    if not request.json:
+        return errors.bad_request("No JSON file detected.")
+
+    if 'done' not in request.json or 'name' not in request.json:
+        return errors.bad_request("The keys 'done' and 'name' not in the JSON")
+
+    bucketlist.name = request.json.get('name')
+    bucketlist.save()
+    return bucketlist, 200
+
+
+@main.route('/bucketlists/<int:list_id>', methods=['DELETE'])
+@auth.login_required
+def delete_bucketlist(list_id):
+    """
+    Delete a BucketList.
+
+    Deletes a BucketList and all items associated with it.
+    """
+    bucketlist = BucketList.query.filter_by(bucketlist_id=list_id).first()
+
+    if not bucketlist or bucketlist.created_by != g.user.user_id:
+        return errors.not_found("The BucketList with the id: {0} doesn't"
+                                " exist.".format(list_id))
+
+    bucketlist.delete()
+    return jsonify({'Delete': True}), 200
+
+
+@main.route(
+    '/bucketlists/<int:list_id>/items/', methods=['POST'], strict_slashes=False
+)
+@auth.login_required
+@json
+def add_bucketlist_item(list_id):
+    """
+    Add new item.
+
+    This function adds a new item to a BucketList. It gets the name and done
+    keys from the json supplied and saves to the database.
+    """
+    bucketlist = BucketList.query.filter_by(bucketlist_id=list_id).first()
+
+    if not bucketlist or bucketlist.created_by != g.user.user_id:
+        return errors.not_found("The BucketList with the id: {0} doesn't"
+                                " exist.".format(list_id))
+
+    if not request.json or 'name' not in request.json:
+        return errors.bad_request("No JSON file detected.")
+
+    item = Items().from_json(request.json)
+    item.bucketlist_id = bucketlist.bucketlist_id
+    item.save()
+    return bucketlist, 201
+
+
+@main.route(
+    '/bucketlists/<int:list_id>/items/<int:item_id>', methods=['PUT']
+)
+@auth.login_required
+@json
+def update_bucketlist_item(list_id, item_id):
+    """
+    Update item.
+
+    This function updates an item in a BucketList.
+    """
+    bucketlist = BucketList.query.filter_by(bucketlist_id=list_id).first()
+
+    if not bucketlist or bucketlist.created_by != g.user.user_id:
+        return errors.not_found("The BucketList with the id: {0} doesn't"
+                                " exist.".format(list_id))
+
+    item = Items.query.get(item_id)
+    if not item or (item.bucketlist_id != bucketlist.bucketlist_id):
+        return errors.not_found("The item with the ID: {0} doesn't exist"
+                                .format(item_id))
+
+    if not request.json or (('name' or 'done') not in request.json):
```

Comment:
This line of code is interesting. To me if feels there is bug somewhere: ('name' or 'done') not in request.json